pylint/pyreverse/writer.py

```diff
@@ -184,7 +253,14 @@ def get_title(self, obj):
         """get project title in vcg format"""
         return r"\fb%s\fn" % obj.title
 
-    def get_values(self, obj):
+    def get_package_properties(self, obj):
+        """get label and shape for packages."""
+        return dict(
```

Comment:
I think it would be better to use ``NamedTuple`` (or a child class) here, so we can have proper typing and not use dict everywhere.

Reply:
I agree that a NamedTuple would be the cleaner solution. However, the _current_ (as in: this PR) implementation of the ``DiagramWriter`` class and implementations of ``DotBackend`` and ``VCGPrinter`` do not allow for this. ``DotBackend`` and ``VCGPrinter`` rely on the data returned by the ``get_package_properties`` and ``get_class_properties`` methods, and the data structures they require are not the same.

That being said: I have already started working on the PlantUML backend, and while doing so I extracted a common interface for the ``Printer`` classes. Using a ``NamedTuple`` should be possible then.

Reply:
I agree with your other comment, it your implementation of plantuml make you upgrade the code with NamedTuple and proper typing it will be nice to rebase on it once it's merged.